packages/amplify-provider-awscloudformation/src/utils/hooks-manager.ts

```diff
@@ -0,0 +1,275 @@
+  /**
+   * uploads all files except ignored files in hooks directory to S3 bucket
+   * returns promise that resolves with list of successfully uploaded files
+   * return void
+   * throws error if upload to S3 fials
+   * deletes the previous state of hooks from S3 completely
+   *
+   * @param {$TSContext} context
+   * @returns {Promise<void>} or throws error
+   */
+  if (skipHooks()) {
+    return;
+  }
+  const hooksDirectoryPath = pathManager.getHooksDirPath(context.exeInfo?.localEnvInfo?.projectPath);
+  await deleteHooksFromS3(context);
+
+  if (!fs.existsSync(hooksDirectoryPath)) {
+    return;
+  }
+
+  const relativeFilePathsToUpload = getNonIgnoredFileList(context);
+  const s3 = await S3.getInstance(context);
+
+  for (const relativeFilePathToUpload of relativeFilePathsToUpload) {
+    const absolutefilePathToUpload = path.join(hooksDirectoryPath, relativeFilePathToUpload);
+    if (fs.existsSync(absolutefilePathToUpload)) {
+      const s3Params = {
+        Body: fs.createReadStream(absolutefilePathToUpload),
+        Key: getS3Key(relativeFilePathToUpload),
+      };
+      try {
+        await s3.uploadFile(s3Params);
+      } catch (ex) {
+        throw ex;
+      }
+    }
+  }
+  return;
+}
+
+export async function downloadHooks(
+  context: $TSContext,
+  backendEnv: { deploymentArtifacts: string },
+  awsConfigInfo: aws.S3.ClientConfiguration,
+): Promise<void> {
+  /**
+   * downloads hooks directory from S3 and places in amplify project.
+   * used when no amplify project exist.
+   *
+   * @param {$TSContext} context
+   * @param {{ deploymentArtifacts: string }} backendEnv backendEnv object used to get deploymentBucket
+   * @param {aws.S3.ClientConfiguratio} awsConfigInfo aws credentials information to create S3 object
+   * @return {Promise<void>}
+   */
+  if (skipHooks()) {
+    return;
+  }
+  if (!backendEnv) {
+    return;
+  }
+  const projectPath = process.cwd();
+  const hooksDirPath = pathManager.getHooksDirPath(projectPath);
+
+  const s3Client = new aws.S3(awsConfigInfo);
+  const deploymentBucketName = backendEnv.deploymentArtifacts;
+
+  const params = {
+    Prefix: S3_HOOKS_DIRECTORY,
+    Bucket: deploymentBucketName,
+  };
+
+  // TODO: logs
+  //   const log = logger("downloadHooks.s3.listObjects", [params]);
+  let listHookObjects;
+  try {
+    // log();
+    listHookObjects = await s3Client.listObjects(params).promise();
+  } catch (ex) {
+    // log(ex);
+    throw ex;
+  }
+
+  // loop over each object in S3 hooks directory and download the file
+  for (const listHookObject of listHookObjects.Contents) {
+    const params = {
+      Key: listHookObject.Key,
+      Bucket: deploymentBucketName,
+    };
+    // TODO: logs
+    // const log = logger('downloadHooks.s3.getObject', [params]);
+    let hooksFileObject = null;
+    try {
+      // log();
+      hooksFileObject = await s3Client.getObject(params).promise();
+    } catch (ex) {
+      // log(ex);
+      throw ex;
+    }
```

Comment:
remove try catch (and anywhere else where the error is just rethrown in the catch)